homeassistant/components/lektrico/sensor.py

```diff
@@ -0,0 +1,294 @@
+            dev_reg = dr.async_get(self._hass)
+            if my_entry.device_id is not None:
+                device = dev_reg.async_get(my_entry.device_id)
+                if device is not None:
+                    dev_reg.async_update_device(device.id, sw_version=data.fw_version)
+        return data
+
+    async def _create_coordinator(self) -> None:
+        """Get the coordinator for a specific device."""
+        coordinator = DataUpdateCoordinator(
+            self._hass,
+            _LOGGER,
+            name=f"{DOMAIN}-{self._name}",
+            update_method=self.async_device_update,
+            # Polling interval. Will only be polled if there are subscribers.
+            update_interval=SCAN_INTERVAL,
+        )
+        await coordinator.async_refresh()
+
+        if not coordinator.last_update_success:
+            raise ConfigEntryNotReady
+
+        self._coordinator = coordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Lektrico charger based on a config entry."""
+    charger: lektricowifi.Charger = hass.data[DOMAIN][entry.entry_id]
+
+    await charger.charger_info()
+    settings = await charger.charger_config()
+    _lektrico_device = LektricoDevice(charger, hass, entry.data[CONF_FRIENDLY_NAME])
+    if not await _lektrico_device.init_device():
+        _LOGGER.error("Error initializing Lektrico Device. Name: 1P7K")
+
+    sensors = []
+
+    sensors.extend(
+        [
+            LektricoSensor(
+                charger,
+                sensor_desc,
+                settings,
+                _lektrico_device,
+                entry.data[CONF_FRIENDLY_NAME],
+            )
+            for sensor_desc in SENSORS
+        ]
+    )
+
+    async_add_entities(sensors, True)
+
+
+class LektricoSensor(CoordinatorEntity, SensorEntity):
+    """The entity class for Lektrico charging stations sensors."""
+
+    entity_description: LektricoSensorEntityDescription
+
+    def __init__(
+        self,
+        charger: lektricowifi.Charger,
+        description: LektricoSensorEntityDescription,
+        settings: lektricowifi.Settings,
+        _lektrico_device: LektricoDevice,
+        friendly_name: str,
+    ) -> None:
+        """Initialize Lektrico charger."""
+        super().__init__(_lektrico_device.coordinator)
+        self.charger = charger
+        self.friendly_name = friendly_name
+        self.entity_description = description
+
+        self._attr_name = f"{self.friendly_name} {description.name}"
+        self._attr_unique_id = f"{settings.serial_number}_{description.name}"
+        # ex: 500006_Led_Brightness
+
+        self._settings = settings
+        self._lektrico_device = _lektrico_device
+
+    @property
+    def native_value(self) -> float | str | Any | None:
+        """Return the state of the sensor."""
+        if self.entity_description.value_fn is not None:
```

Comment:
Make `value_fn` required in the entity description, so we don't need this check anymore.

Reply:
I changed it, thank you.